system/lib/wasmfs/file.h

```diff
@@ -20,6 +20,43 @@
 
 namespace wasmfs {
 
+class path_string {
```

Comment:
Let's name this `PathString` to match the convention of using PascalCase for user types. It would also be good to add a comment explaining that it exists solely to support case insensitivity.